runtime/src/accounts_db.rs

```diff
@@ -3660,79 +3656,205 @@ impl AccountsDB {
         F: Fn(LoadedAccount, &mut B, Slot) + Send + Sync,
         B: Send + Default,
     {
-        snapshot_storages
-            .into_par_iter()
-            .flatten()
-            .map(|storage| {
-                let accounts = storage.accounts.accounts(0);
+        let items: Vec<_> = snapshot_storages.iter().flatten().collect();
+
+        // Without chunks, we end up with 1 output vec for each outer snapshot storage.
+        // This results in too many vectors to be efficient.
+        const MAX_ITEMS_PER_CHUNK: usize = 5_000;
```

Comment:
Should this be calculated based on `snapshot_storages.len()` and `num_cpus::get()`?

Reply:
I'm not sure on the heuristics of getting a number here. I know no number is not good. I just picked something and it had reasonable performance so I haven't gone back to tune this yet. I assumed as I/we learned something more about the dynamics of the system, more intuitive understanding may appear. I have no idea how cpu or io bound these functions are.
I don't like it being a magic number, so at least its a constant, but I'm not sure what the right answer is yet.
I don't mean to sound defensive in these responses, fwiw.